### PR TITLE
Add tests for language files

### PR DESCRIPTION
**Pull request by MichaelZetune**
### Description

This PR adds some tests to enforce the following invariants on the language files:
* All keys in `messages` must be present in all other messages files
* An `IGNORE_LIST` is defined to enumerate exceptions to this rule. If a key is on the ignore list, (1) it must be in `messages` and (2) *may or may not* not be in one or more other messages files.

In a future PR:
* We will enforce that the keys in `messages` and other messages files must be the **exact same** (except for the `IGNORE_LIST`), as opposed to the current state where the other language files just need to *have* those same keys. Once this is complete, if a key is on `IGNORE_LIST` it **may not** be in any language file other than `messages`.
   * Currently, we cannot do this because there are keys that are present in some but not all non-primary language files.

### Checklist

- [x] Created `test/conf/MessagesTest.java` as per description above.
- [x] Delete `bin/list-incomplete-messages` which does the same thing as this test (but doesn't run on CI)

## Issues

Progress towards #4520.

### CODE REVIEW

File: bin/list-incomplete-messages
Comment (bion) [inline bin/list-incomplete-messages:1]: **issue (blocking):** It seems like this still provides value the java test doesn't yet: knowing which languages a given key is missing?

File: server/test/conf/MessagesTest.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+import org.junit.Test;
+
+public class MessagesTest {
+
+  // The file path of the primary language file.
+  String PRIMARY_LANGUAGE_FILE = "conf/messages";
+
+  // The file paths of all non-primary language files, including `en-US`.
+  private static final Set<String> OTHER_LANGUAGE_FILES =
+      Set.of(
```
Comment (bion) [inline server/test/conf/MessagesTest.java:20]: **nitpick:** prefer immutable collections
```suggestion
      ImmutableSet.of(
```

File: server/test/conf/MessagesTest.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+          "conf/messages.am",
+          "conf/messages.en-US",
+          "conf/messages.es-US",
+          "conf/messages.ko",
+          "conf/messages.so",
+          "conf/messages.tl",
+          "conf/messages.vi",
+          "conf/messages.zh-TW");
```
Comment (bion) [inline server/test/conf/MessagesTest.java:28]: **suggestion (if-minor, non-blocking):** could glob here instead `conf/messages.*`

File: server/test/conf/MessagesTest.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+import java.util.Properties;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.stream.Collectors;
+import org.junit.Test;
+
+public class MessagesTest {
+
+  // The file path of the primary language file.
+  String PRIMARY_LANGUAGE_FILE = "conf/messages";
```
Comment (bion) [inline server/test/conf/MessagesTest.java:16]: **nitpick:** 
```suggestion
  private static final String PRIMARY_LANGUAGE_FILE = "conf/messages";
```

File: server/test/conf/MessagesTest.java [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    // and are merged.
+    keysInPrimaryFile.removeAll(IGNORE_LIST);
+
+    for (String file : OTHER_LANGUAGE_FILES) {
+      TreeSet<String> keysInForeignLangFile = keysInFile(file);
+
+      // Checks that the language file contains exactly the same message keys as the
+      // primary language file, *in the same order*.
+      //
+      // TODO(MichaelZetune): change containsSubsequence to containsExactlyElementsOf.
```
Comment (bion) [inline server/test/conf/MessagesTest.java:72]: **nitpick:** reference an issue number please

**Review by bion — CHANGES_REQUESTED** [review]

File: server/test/conf/MessagesTest.java [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+
+import java.io.FileInputStream;
+import java.io.InputStream;
+import java.util.Properties;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.stream.Collectors;
+import org.junit.Test;
+
+public class MessagesTest {
```
[the remark this answers is not in the document]
Comment (MichaelZetune) [reply server/test/conf/MessagesTest.java:13]: Done.

File: server/test/conf/MessagesTest.java (hunk 3 above)
Comment (MichaelZetune) [reply server/test/conf/MessagesTest.java:16]: Done.

File: server/test/conf/MessagesTest.java (hunk 1 above)
Comment (MichaelZetune) [reply server/test/conf/MessagesTest.java:20]: Done.

File: server/test/conf/MessagesTest.java (hunk 4 above)
Comment (MichaelZetune) [reply server/test/conf/MessagesTest.java:72]: Done.

File: server/test/conf/MessagesTest.java [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+    for (String file : OTHER_LANGUAGE_FILES) {
+      TreeSet<String> keysInForeignLangFile = keysInFile(file);
+
+      // Checks that the language file contains exactly the same message keys as the
+      // primary language file, *in the same order*.
+      //
+      // TODO(MichaelZetune): change containsSubsequence to containsExactlyElementsOf.
+      //  Currently this is not possible, even with IGNORE_LIST, because there are keys
+      //  that are present in some but not all of the non-primary language files.
+      assertThat(keysInForeignLangFile).containsSubsequence(keysInPrimaryFile);
```
[the remark this answers is not in the document]
Comment (MichaelZetune) [reply server/test/conf/MessagesTest.java:75]: Done, now it says a message like:

```
[error] Test conf.MessagesTest.testMessages failed: java.lang.AssertionError: [content.mobileFileUploadHelp, content.requiredFieldsAnnotation] found in primary language file but not in conf/messages.ko. Add these keys to conf/messages.ko to resolve this issue., took 0.218 sec
[error]     at conf.MessagesTest.testMessages(MessagesTest.java:86)
[error]     at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[error]     at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[error]     at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[error]     at java.lang.reflect.Method.invoke(Method.java:566)
[error]     ...
```

File: server/test/conf/MessagesTest.java (hunk 2 above)
Comment (MichaelZetune) [reply server/test/conf/MessagesTest.java:28]: Done.

File: bin/list-incomplete-messages
Comment (MichaelZetune) [reply bin/list-incomplete-messages:1]: I believe I've added this value to the Java test now; let me know if you disagree.

Comment (MichaelZetune) [reply bin/list-incomplete-messages:1]: Edit: well, you can't *give* the test a key and ask what files are missing it, but it's pretty close with the error message. Let me know if that's not close enough and you prefer to keep the script.

Comment (MichaelZetune) [reply bin/list-incomplete-messages:1]: I've updated the test to use `@Parameters`, so we run one test per file. Now we get output like this (one key missing from one file and another key missing from another file):

```
[error] Test conf.MessagesTest.messages_keysInPrimaryFileInAllOtherFiles(conf/messages.es-US) [2] failed: java.lang.AssertionError: [button.deleteFile] found in primary language file but not in conf/messages.es-US. Add these keys to conf/messages.es-US or to the ignore list in conf.MessagesTest to resolve this issue., took 0.008 sec
[error]     at conf.MessagesTest.messages_keysInPrimaryFileInAllOtherFiles(MessagesTest.java:85)
[error]     at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[error]     at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[error]     at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[error]     at java.lang.reflect.Method.invoke(Method.java:566)
[error]     ...
[info] Test conf.MessagesTest.messages_keysInPrimaryFileInAllOtherFiles(conf/messages.ko) [3] started
[info] Test conf.MessagesTest.messages_keysInPrimaryFileInAllOtherFiles(conf/messages.so) [4] started
[error] Test conf.MessagesTest.messages_keysInPrimaryFileInAllOtherFiles(conf/messages.so) [4] failed: java.lang.AssertionError: [content.or] found in primary language file but not in conf/messages.so. Add these keys to conf/messages.so or to the ignore list in conf.MessagesTest to resolve this issue., took 0.002 sec
[error]     at conf.MessagesTest.messages_keysInPrimaryFileInAllOtherFiles(MessagesTest.java:85)
[error]     at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[error]     at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[error]     at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[error]     at java.lang.reflect.Method.invoke(Method.java:566)
[error]     ...
```

**Review by bion — APPROVED** [review]

**Pull request merged**